### PR TITLE
Upgrade build target to es6

### PR DESCRIPTION
**Pull request by nickpape-msft**
Note, this means users of our code will need to be using >= Node 6 (which is the current LTS version)

### CODE REVIEW

File: core-build/gulp-core-build/src/tasks/GulpTask.ts [hunk 1]
```diff
@@ -277,11 +277,9 @@ export abstract class GulpTask<TTaskConfig> implements IExecutable {
             (file: gutil.File,
               encoding: string,
               callback: (p?: Object) => void) => {
-                'use strict';
```
Comment (qz2017) [inline core-build/gulp-core-build/src/tasks/GulpTask.ts]: >'use strict'; [](start = 16, length = 13)

why do you remove it?

File: core-build/gulp-core-build/tsconfig.json [hunk 2]
```diff
@@ -17,6 +17,7 @@
       "es5",
       "scripthost",
       "es2015.promise",
+      "es2015.iterable",
```
Comment (qz2017) [inline core-build/gulp-core-build/tsconfig.json]: >"es2015.iterable", [](start = 6, length = 18)

This package does not appear to need it.

File: core-build/gulp-core-build-karma/tsconfig.json [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
     "jsx": "react",
     "declaration": true,
     "sourceMap": true,
     "strictNullChecks": true,
     "types": [
       "node"
+    ],
+    "lib": [
+      "es5",
+      "es2015.iterable"
```
Comment (qz2017) [inline core-build/gulp-core-build-karma/tsconfig.json]: Does this package need those libraries?

File: core-build/gulp-core-build/src/tasks/GulpTask.ts (hunk 1 above)
Comment (nickpape-msft) [reply core-build/gulp-core-build/src/tasks/GulpTask.ts]: It throws an exception. For example, you can't use `...` or default values in functions if you add "use strict"

File: core-build/gulp-core-build/tsconfig.json (hunk 2 above)
Comment (nickpape-msft) [reply core-build/gulp-core-build/tsconfig.json]: I made sure that nothing had it, but then I added it back only if the build broke without having it... I think something with the `"target": "es6"` vs `"lib": "es5"` causes it.

File: core-build/gulp-core-build-karma/tsconfig.json (hunk 3 above)
Comment (nickpape-msft) [reply core-build/gulp-core-build-karma/tsconfig.json]: Yeah, I was surprised, but I needed es2015.iterable in a lot more places than I thought.. If I remove it you get an error like this:

```
 typescript - unknown filename(0,0): error TS2318: Cannot find global type 'Symbol'.
```

File: libraries/decorators/tsconfig.json [hunk 4]
```diff
@@ -1,14 +1,19 @@
 {
   "compilerOptions": {
-    "target": "es5",
+    "target": "es6",
```
Comment (qz2017) [inline libraries/decorators/tsconfig.json]: >es6 [](start = 15, length = 3)

this is not just for tools.

Comment (nickpape-msft) [reply libraries/decorators/tsconfig.json]: Oh ok

File: libraries/load-themed-styles/tsconfig.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "target": "es5",
```
Comment (qz2017) [inline libraries/load-themed-styles/tsconfig.json]: >es5 [](start = 15, length = 3)

this as well.

**Pull request merged**